lib/view_component/compiler.rb

```diff
@@ -265,12 +292,16 @@ def compiled_template(file_path)
       end
     end
 
-    def call_method_name(variant)
-      if variant.present? && variants.include?(variant)
-        "call_#{variant}"
-      else
-        "call"
-      end
+    def call_method_name(template, variant)
+      name =
+        if template.blank? || template == component_class.name.demodulize.underscore
+          "call"
+        else
+          "render"
+        end
+      name += "_#{variant}" if variant.present? && variants.include?(variant)
```

Comment:
Might we add a test for sub-template variants?

Reply:
Hmm, this comment brings up an interesting point. The API effectively outsources variant handling to the user, because there is no dispatch to the correct variant automatically. That is, I would need to do something like:

```ruby
if variant == 'some_variant'
  render_list_template_some_variant
else
  render_list_template
end
```

This is not very user-friendly. I'm wondering what a better API would look like? Maybe the compiler should compile to `__render_<foo>_template` (that is, add a `__` prefix), and then `render_<foo>_template` would do the variant checking?

```ruby
def render_list_template elements:
   case @__vc_variant
   when 'some_variant'
      __render_list_template_some_variant elements: elements
   else
      __render_list_template elements: elements
   end
end
```

Reply:
Hmm. What happens if the variants have different Strict Locals?

Reply:
That sounds unusual, but we could use argument forwarding (`**args`) to support that

Reply:
I would prefer that we just generate a method for each variant. It's simpler. We already do this for the `call_` methods that can be defined by a consumer.